packages/react/src/components/ActionMenu/ActionMenu.tsx

```diff
@@ -31,6 +33,12 @@ type ActionMenuProps = {
   trigger: React.ReactElement | ActionMenuTriggerFunction;
   /** Render the action menu in a different location in the dom. */
   portal?: React.RefObject<HTMLElement> | HTMLElement;
+  /**
+   * If true, the menu will will be passed as a children prop to the trigger function.
+   * If false, the menu will render as a sibling of the trigger.
+   * Only supported if trigger is a function *and* portal is undefined.
+   */
+  renderInTrigger?: boolean;
```

Comment:
This prop feels a bit janky to me; @scurker , if you have a better proposal for the name, or if you'd prefer to infer this based on whether the menu is within a top bar somehow, I'd be happy to update accordingly.

Reply:
Yeah I agree (see linked comment). If we can easily infer that its inside of a MenuBar, that would be much cleaner.

Reply:
I gave this some thought and experimenting but didn't come up with anything that felt significantly better - TopBar/MenuBar don't provide any sort of context for us to look for already and it felt wrong to add it just as a marker. It felt like the better of bad options to expose the prop; I don't think it's *totally* unreasonable for ActionMenu to allow the consumer control over this decision.